MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,25 +26,91 @@ import Foundation
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { [weak self] result, error in
+            guard let self = self else { return }
+
+            if let error = error as? NSError {
+                let accessTokenError = self.createRetrieveAccessTokenError(error: error, context: context)
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? context.correlationId())) }
         }
     }
+    
+    func createRetrieveAccessTokenError(error: NSError, context: MSALNativeAuthRequestContext) -> RetrieveAccessTokenError {
+        if let innerError = error.userInfo[NSUnderlyingErrorKey] as? NSError,
+           let message = innerError.userInfo[MSALErrorDescriptionKey] as? String,
+           let code = codeAndCorrelationIdFromMSALError(error: innerError).code,
+           let correlationId = codeAndCorrelationIdFromMSALError(error: innerError).correlationId {
+            return RetrieveAccessTokenError(type: .generalError, message: message, correlationId: correlationId, errorCodes: [code])
+        }
+        
+        if let message = error.userInfo[MSALErrorDescriptionKey] as? String,
+            let code = codeAndCorrelationIdFromMSALError(error: error).code,
+            let correlationId = codeAndCorrelationIdFromMSALError(error: error).correlationId {
+            return RetrieveAccessTokenError(type: .generalError, message: message, correlationId: correlationId, errorCodes: [code])
+        }
+        
+        return RetrieveAccessTokenError(type: .generalError,
+                                        message: error.localizedDescription,
+                                        correlationId: context.correlationId(),
+                                        errorCodes: [error.code])
+    }
+
+    private func codeAndCorrelationIdFromMSALError(error: NSError) -> (code: Int?, correlationId: UUID?) {
+        var errorCode: Int?
```

Comment:
nit: this method can be shortened with:
```suggestion
        let errorCode = error.userInfo[MSALInternalErrorCodeKey] as? Int
        let errorCorrelationId = UUID(uuidString: error.userInfo[MSALCorrelationIDKey] as? String ?? "")
        return (errorCode, errorCorrelationId)
```

Reply:
Done.